elytron/WFLY-14984-encrypted-expression-for-client-config.adoc

```diff
@@ -0,0 +1,172 @@
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14984[WFLY-14984]
+
+// === Related Issues
+
+// * https://issues.redhat.com/browse/WFCORE[WFCORE-XXXX]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* The ability to specify attribute values as encrypted expressions should be added to wildfly-config.xml. 
+
+* The parsing needs to be able to run separately from the application server, as the server may not be running while the client configuration is being set up. 
+
+* Configurations may not depend on components created for server configuration, such as `resolvers` or `elytron/expression`. The resolver for the encrypted expression should be configured through the client configuration. 
+
+* The `encrypted-expression` would be a new configuration block added to the client config under the new schema. The new attributes added include `resolvers`, with attributes called `resolver` under it. Additionally, <credential-stores> attribute would be added to the schema with the <credential-store> attributes added to it. The structures of this attribute would be the same as the one one under the `authentication-client` schema. 
+
+** There may be multiple `resolver` attribute listed under <resolvers>, and there may be multiple <credential-store> attributes added under <credential-stores>.
+
+** The configuration for this block can be specified as follows: 
+```
+<encrypted-expression>
+	<credential-store name="credential-store-name" type="PKCS12" provider="provider-name"/>
+        <attributes>
+            <attribute name="path" value="path/to/credential-store-name.cs" />
+        </attributes>
+        <protection-parameter-credentials>
+            <clear-password password="password" />
+        </protection-parameter-credentials>
+    </credential-store>
+    <resolvers>
+        <resolver name= “resolver-name” credential-store= “credential-store-name key=“key”/>
```

Comment:
credential-store-name is missing the closing quote here and below.

Reply:
fixed